### PR TITLE
[libarchive, libuv]Fix static linkage

### PR DESCRIPTION
**Pull request by Neumann-A**
adds a wrapper to correctly link in dependencies for libarchive and libuv

### CODE REVIEW

File: ports/libarchive/portfile.cmake [hunk 1]
```diff
@@ -38,15 +55,26 @@ vcpkg_configure_cmake(
...
+    string(REPLACE "OFF" "0" _feature "${_feature}")
+    list(GET _feature 0 _feature_name)
+    list(GET _feature 1 _feature_status)
+    message(STATUS ${_feature_name} status ${_feature_status})
+    set(${_feature_name} ${_feature_status})
+endforeach()
+configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" "${CURRENT_PACKAGES_DIR}/share/${PORT}/vcpkg-cmake-wrapper.cmake" @ONLY)
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 foreach(HEADER ${CURRENT_PACKAGES_DIR}/include/archive.h ${CURRENT_PACKAGES_DIR}/include/archive_entry.h)
     file(READ ${HEADER} CONTENTS)
```
Comment (LilyWangL) [inline ports/libarchive/portfile.cmake:80]: Can you please use command `file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

File: ports/libuv/portfile.cmake [hunk 2]
```diff
@@ -32,8 +32,10 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
     string(REPLACE "defined(USING_UV_SHARED)" "1" UV_H "${UV_H}")
 else()
     string(REPLACE "defined(USING_UV_SHARED)" "0" UV_H "${UV_H}")
+    configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" "${CURRENT_PACKAGES_DIR}/share/${PORT}/vcpkg-cmake-wrapper.cmake" @ONLY)
 endif()
 file(WRITE ${CURRENT_PACKAGES_DIR}/include/uv.h "${UV_H}")
 
 file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libuv)
 file(RENAME ${CURRENT_PACKAGES_DIR}/share/libuv/LICENSE ${CURRENT_PACKAGES_DIR}/share/libuv/copyright)
+
```
Comment (LilyWangL) [inline ports/libuv/portfile.cmake:41]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Comment by LilyWangL** [comment]
`include(vcpkg_common_functions)` is deprecated in _libuv/portfile.cmake_. Please remove it.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/libarchive/portfile.cmake [hunk 3]
```diff
@@ -38,21 +55,31 @@ vcpkg_configure_cmake(
         -DENABLE_CAT=OFF
         -DENABLE_XATTR=OFF
         -DENABLE_ACL=OFF
-        -DENABLE_TEST=OFF
         -DENABLE_ICONV=OFF
+        -DENABLE_TEST=OFF
         -DPOSIX_REGEX_LIB=NONE
```
Comment (ras0219-msft) [inline ports/libarchive/portfile.cmake:60]: Should this option be removed since it is specified above?

File: ports/libarchive/portfile.cmake [hunk 4]
```diff
@@ -38,21 +55,31 @@ vcpkg_configure_cmake(
...
 vcpkg_copy_pdbs()
 
+foreach(_feature IN LISTS FEATURE_OPTIONS)
+    string(REPLACE "-D" "" _feature "${_feature}")
+    string(REPLACE "=" ";" _feature "${_feature}")
+    string(REPLACE "ON" "1" _feature "${_feature}")
+    string(REPLACE "OFF" "0" _feature "${_feature}")
+    list(GET _feature 0 _feature_name)
+    list(GET _feature 1 _feature_status)
+    message(STATUS ${_feature_name} status ${_feature_status})
```
Comment (ras0219-msft) [inline ports/libarchive/portfile.cmake:74]: Please remove this debugging message

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by ras0219-msft** [comment]
LGTM, thanks!

**Pull request merged**